### PR TITLE
Add functionality to comments

### PR DESCRIPTION
**Pull request by Jamlons**
- Added further styling to comments and replies
- Added proper comment error flow
- Comments will now auto-refresh from the database. Checking if the current comment list is up to date
- Comments are filtered best to worse
- Submit buttons now feel more responsive.

### CODE REVIEW

File: app/main/routes.py [hunk 1]
```diff
@@ -28,48 +28,83 @@ def index():
...
 
     # Handle pagination and query for posts as usual
     page = request.args.get('page', 1, type=int)
-    posts = db.paginate(query, page=page, per_page=current_app.config['POSTS_PER_PAGE'], error_out=False)
+
+    # Re-add filter argument if it exists
+    if 'filter' in request.args:
+        posts = db.paginate(query, page=page, per_page=current_app.config['POSTS_PER_PAGE'], error_out=False)
+    else:
+        posts = db.paginate(query, page=page, per_page=current_app.config['POSTS_PER_PAGE'], error_out=False)
```
Comment (Skoll-8) [inline app/main/routes.py:71]: Line 71 and Line 69 are the same, making this 'if' statement arbitrary.

File: app/static/index.js [hunk 2]
```diff
@@ -84,6 +84,10 @@ function buttonHandling() {
                 button.classList.remove("fa-regular");
                 button.classList.add("fa-solid");
             }
+            else if (isClicked) {
```
Comment (Skoll-8) [inline app/static/index.js:87]: In my opinion, the comment button feels better if it stays darkened on mouse hover if it's been clicked.

Comment (Skoll-8) [reply app/static/index.js:87]: So I think it will feel better without this 'if' statement

**Review by Skoll-8 — APPROVED** [review]

File: app/static/index.js (hunk 2 above)
Comment (Jamlons) [reply app/static/index.js:87]: Lets do a poll - I prefer the if statement handling

**Pull request merged**